src/lib.rs

```diff
@@ -0,0 +1,458 @@
+
+                            Ok(run_check_release(
+                                &mut config,
+                                crate_name,
+                                current_crate,
+                                baseline_crate,
+                            )?)
+                        }
+                    })
+                    .collect()
+            }
+        };
+        let success = all_outcomes
+            .into_iter()
+            .fold_ok(true, std::ops::BitAnd::bitand)?;
+
+        Ok(Report { success })
+    }
+}
+
+#[non_exhaustive]
+#[derive(Debug)]
+pub struct Report {
+    success: bool,
+}
+
+impl Report {
+    pub fn success(&self) -> bool {
+        self.success
+    }
+}
+
+fn generate_versioned_crates(
+    config: &mut GlobalConfig,
+    rustdoc_cmd: &RustdocCommand,
+    current_loader: &dyn rustdoc_gen::RustdocGenerator,
+    baseline_loader: &dyn rustdoc_gen::RustdocGenerator,
+    crate_name: &str,
+    version: Option<&Version>,
+) -> anyhow::Result<(VersionedCrate, VersionedCrate)> {
+    let current_path = current_loader.load_rustdoc(
+        config,
+        rustdoc_cmd,
+        rustdoc_gen::CrateDataForRustdoc {
+            name: crate_name,
+            crate_type: rustdoc_gen::CrateType::Current,
+        },
+    )?;
+    let current_crate = load_rustdoc(&current_path)?;
+
+    // The process of generating baseline rustdoc can overwrite
+    // the already-generated rustdoc of the current crate.
+    // For example, this happens when target-dir is specified in `.cargo/config.toml`.
+    // That's the reason why we're immediately loading the rustdocs into memory.
+    // See: https://github.com/obi1kenobi/cargo-semver-checks/issues/269
+    let baseline_path = baseline_loader.load_rustdoc(
+        config,
+        rustdoc_cmd,
+        rustdoc_gen::CrateDataForRustdoc {
+            name: crate_name,
+            crate_type: rustdoc_gen::CrateType::Baseline {
+                highest_allowed_version: version,
+            },
+        },
+    )?;
+    let baseline_crate = load_rustdoc(&baseline_path)?;
+
+    Ok((current_crate, baseline_crate))
+}
+
+fn manifest_from_dir(manifest_dir: &Path) -> PathBuf {
+    manifest_dir.join("Cargo.toml")
+}
+
+fn manifest_metadata(manifest_dir: &Path) -> anyhow::Result<cargo_metadata::Metadata> {
+    let manifest_path = manifest_from_dir(manifest_dir);
+    let mut command = cargo_metadata::MetadataCommand::new();
+    let metadata = command.manifest_path(manifest_path).exec()?;
+    Ok(metadata)
+}
+
+fn manifest_metadata_no_deps(manifest_dir: &Path) -> anyhow::Result<cargo_metadata::Metadata> {
+    let manifest_path = manifest_from_dir(manifest_dir);
+    let mut command = cargo_metadata::MetadataCommand::new();
+    let metadata = command.manifest_path(manifest_path).no_deps().exec()?;
+    Ok(metadata)
+}
+
+fn get_target_dir_from_config() -> Option<PathBuf> {
+    // TODO: I'll implement this function probably tomorrow
+    None
```

Comment:
As the comment says, I'll come back to this function in a few days.

Reply:
We discussed on Zulip whether this is the best idea and we've decided to instead rely on the XDG cache directory: 2449acd377ed1792127a10c0ffed211094664d1b